cni/pkg/nodeagent/podcgroupns.go

```diff
@@ -119,10 +132,27 @@ func (p *PodNetnsProcFinder) processEntry(proc fs.FS, netnsObserved sets.Set[uin
 		return nil, err
 	}
 
-	inode, err := GetInode(fi)
+	// Check the starttime of the proc in question,
+	// in case there are multiple procs with different netnamespaces
+	// in the same pod - we want the oldest in all cases.
+	ownerProcStarttime, err := GetStarttime(proc, entry)
 	if err != nil {
 		return nil, err
 	}
+
+	inode, dev, err := GetInodeDev(fi)
+	if err != nil {
+		return nil, err
+	}
+
+	// It is possible (but unlikely, see https://github.com/istio/istio/issues/55139) that we may get a pod netns
+	// that is == the hostnetns. This might lead to us breaking the host, so ignore everything that looks like
+	// the host netns.
+	if host, err := isHostNetns(proc, inode, dev); host || err != nil {
+		log.Warnf("netns: ignoring host netns (ino %d and dev %d) %s", inode, dev, err)
```

Comment:
hmm, wondering if we should have this at warn? Given in some envs its going to happen every single time

Reply:
err is error, should be `%v`

Reply:
I think even at debug level we need to move this to only warn if its matching the pod. Otherwise we are going to run this against the entire process tree and there are a ton of processes in the host network